unit-tests-java8/src/test/java/org/eclipse/collections/test/map/mutable/MutableMapIterableTestCase.java

```diff
@@ -126,6 +127,28 @@ default void Map_entrySet_remove()
         }
     }
 
+    @Test
+    default void MutableMapIterable_removeIf()
+    {
+        MutableMapIterable<Integer, String> map1 = this.newWithKeysValues(1, "1", 2, "Two", 3, "Three");
+
+        assertFalse(map1.removeIf(Predicates2.alwaysFalse()));
+        assertEquals(this.newWithKeysValues(1, "1", 2, "Two", 3, "Three"), map1);
+        assertTrue(map1.removeIf(Predicates2.alwaysTrue()));
+        assertEquals(this.newWithKeysValues(), map1);
+
+        MutableMapIterable<Integer, String> map2 = this.newWithKeysValues(1, "One", 2, "TWO", 3, "THREE", 4, "four", 5, "Five", 6, "Six", 7, "Seven", 8, "Eight");
+        assertTrue(map2.removeIf((each, value) -> each % 2 == 0 && value.length() < 4));
+        assertEquals(this.newWithKeysValues(1, "One", 3, "THREE", 4, "four", 5, "Five", 7, "Seven", 8, "Eight"), map2);
+
+        assertTrue(map2.removeIf((each, value) -> each % 2 != 0 && value.equals("THREE")));
+        assertEquals(this.newWithKeysValues(1, "One", 4, "four", 5, "Five", 7, "Seven", 8, "Eight"), map2);
+
+        assertTrue(map2.removeIf((each, value) -> each % 2 != 0));
+        assertFalse(map2.removeIf((each, value) -> each % 2 != 0));
+        assertEquals(this.newWithKeysValues(4, "four", 8, "Eight"), map2);
```

Comment:
The important changes are:
* I pulled the test up to the matching interface. MutableMapIterable.removeIf should be tested in MutableMapIterableTestCase.
* I create expected maps using newWithKeysValue, which creates them in order.
* I use IterableTestCase.assertEquals, not Assert.assertEquals or methods on Verify. This asserts ordering.